src/views/tutorial-view/server.ts

```diff
@@ -137,44 +143,47 @@ const moizeOpts: Options = {
 // limit the expensive call for tutorials that have the same product
 const cachedGetAllCollections = moize(getAllCollections, moizeOpts)
 
-export async function getTutorialPagePaths(
-	product: ProductOption
-): Promise<TutorialPagePaths[]> {
-	const allCollections = await cachedGetAllCollections({
-		product: { slug: product },
-	})
-	// Only build collections where this product is the main 'theme'
+export async function getTutorialPagePaths(): Promise<TutorialPagePaths[]> {
+	const allCollections = await cachedGetAllCollections()
+	let paths = []
+	// Only build collections where the `theme` is a valid beta product
 	// @TODO once we implement the `theme` query option, remove the theme filtering
 	// https://app.asana.com/0/1201903760348480/1201932088801131/f
-	const filteredCollections = allCollections.filter((c) => c.theme === product)
+	const filteredCollections = allCollections.filter((c) =>
+		getIsBetaProduct(c.theme as LearnProductSlug)
+	)
 	// go through all collections, get the collection slug
-	const paths = filteredCollections.flatMap((collection) => {
-		const collectionSlug = splitProductFromFilename(collection.slug)
+	const currentProductPaths = filteredCollections.flatMap((collection) => {
+		// assuming slug structure of :product/:filename
+		const [productFromCollection, collectionSlug] = collection.slug.split('/')
 		// go through the tutorials within this collection, create a path for each
 		return collection.tutorials.map((tutorial) => {
 			const tutorialSlug = splitProductFromFilename(tutorial.slug)
 
 			return {
 				params: {
+					product: productFromCollection,
 					tutorialSlug: [collectionSlug, tutorialSlug] as [string, string],
 				},
 			}
 		})
 	})
+	paths = [...paths, ...currentProductPaths]
```

Comment:
📌 Same future-thoughts here, I think we can simplify this with a `allCollections.forEach` that pushes to the `paths` array. 💭